### ISSUE TITLE
Text-to-CAD preview pane should share the configured mouse controls from the modeling pane

### ISSUE THREAD

**Issue by greg-kcio** [KittyCAD/modeling-app#3463]

**Comment by franknoirot** [comment]
Okay, so after a bit of investigation I've determined that the THREE.js OrbitControls do not support our kind of implementation for pointer events. It has its opinions about the modifier key behavior baked into the implementation, and only exposed the ability for the consumer to switch off one of the interaction types (pan, rotate, or zoom), or reconfigure their mapping to individual buttons with their [`mouseButtons` interface](https://threejs.org/docs/#examples/en/controls/OrbitControls.mouseButtons).

There are good lessons, however, in [their implementation of the OrbitControls class](https://github.com/mrdoob/three.js/blob/master/examples/jsm/controls/OrbitControls.js#L211), which is nicely written and not too hard to follow. We have talked elsewhere about making our control scheme more touch-friendly, and the OrbitControls class encapsulates that logic alongside mouse logic well.

I think we should make an OrbitControls class of our own that is maybe a little slimmer than THREE.js's, and use it both for controlling our stream orbit controls and this Text-to-CAD canvas (perhaps by providing a wrapper that makes our base class THREE.js-ready?).

I consider that more of a refactor than a bug fix, so I'm gonna take just a little more time to see if I can get a quick win here before moving on.

**Comment by franknoirot** [comment]
No, I don't think there is a quick win version of fixing this.

I was able to hack together something approximating the behavior [here](https://github.com/KittyCAD/modeling-app/commit/c03357d8972f4d640a8fe1240eb2acbba6492e3e#diff-6eee61619cce619c03bc549302312fe39262ff06b6a758c1b811a2d77f502e65R187-R237). However, [this line](https://github.com/mrdoob/three.js/blob/master/examples/jsm/controls/OrbitControls.js#L1261) shows just how opinionated the THREE.js OrbitControls class is. No matter what I do to tell it from the outside "do this action on mousemove", there is still logic for keydown handling right at the heart of it. So holding down the <kbd>Shift</kbd>, <kbd>Control</kbd>, or <kbd>Meta</kbd> keys when trying to orbit will force it to pan, and visa versa. [Here](https://github.com/mrdoob/three.js/blob/master/examples/jsm/controls/OrbitControls.js#L17-L19) are the control scheme opinions laid out in the OrbitControls as a doc comment.

I'm removing myself and moving on for now.

**Comment by franknoirot** [comment]
This isn't "blocked" by another external issue, it's just a high-effort low reward since at the moment most users don't even know that's a 3D scene, or even that we have a text-to-cad feature in the app.

The next step for this will be to try to write our own implementation of OrbitControls that honors our interaction guard system, perhaps borrowing from our CameraControls class (which was itself based on OrbitControls originally). Adapting CameraControls to not need to send to the engine, manage the clientSideScene, etc sounds like too much effort, but if splitting out a smaller class that just manages the interaction listeners is straightforward maybe we should do that, and then use that smaller class both in CameraControls and for this new use case.

**Comment by franknoirot** [comment]
As stated earlier, I think this is more of a feature request than a bug fix, as users of all platforms will have an easy enough time figuring out how to orbit the scene even if they're thrown off that the controls are different. Recategorizing

**Comment by greg-motif** [comment]
The Text-to-CAD pane has been replaced with Zookeeper
